File: TaskC.py
import sys
import getpass

def input_password(prompt='Password: '):
    import sys
    import tty
    import termios
  
    sys.stdout.write(prompt)
    sys.stdout.flush()
    password = ""
    fd = sys.stdin.fileno()
    old_settings = termios.tcgetattr(fd)
    try:
        tty.setraw(fd)
        while True:
            ch = sys.stdin.read(1)
            if ch == '\n' or ch == '\r':
                print()
                break
            elif ch == '\x7f':  
                if len(password) > 0:
                    password = password[:-1]
                    sys.stdout.write('\b \b')
                    sys.stdout.flush()
            else:
                password += ch
                sys.stdout.write('*')
                sys.stdout.flush()
    finally:
        termios.tcsetattr(fd, termios.TCSADRAIN, old_settings)
    return password
  
correct_password = "default123"

while True:
    user_input = input_password("Enter password: ")
    if user_input == correct_password:
        print("Access granted")
        break
    else:
        print("Incorrect password, try again.")

File: TaskC.c
#include <stdio.h>
#include <termios.h>
#include <unistd.h>
#include <string.h>

void getPassword(char *password, int maxLength) {
    struct termios oldt, newt;
    int i = 0;
    char ch;

    tcgetattr(STDIN_FILENO, &oldt);
    newt = oldt;
    newt.c_lflag &= ~(ECHO | ICANON);
    tcsetattr(STDIN_FILENO, TCSANOW, &newt);

    while ((read(STDIN_FILENO, &ch, 1) == 1) && ch != '\n' && i < maxLength - 1) {
        if (ch == 127 || ch == 8) {  
            if (i > 0) {
                i--;
                printf("\b \b");
            }
        } else {
            password[i++] = ch;
            printf("*");
        }
        fflush(stdout);
    }
    password[i] = '\0';

    tcsetattr(STDIN_FILENO, TCSANOW, &oldt);
    printf("\n");
}

int main() {
    char password[100];

    while (1) {
        printf("Enter password: ");
        fflush(stdout);
        getPassword(password, sizeof(password));

        if (strcmp(password, "default123") == 0) {
            printf("Access granted\n");
            break;
        } else {
            printf("Incorrect password, try again.\n");
        }
    }

    return 0;
}
